src/Api/ErrorParser/JsonParserTrait.php

```diff
@@ -15,10 +15,14 @@ trait JsonParserTrait
     private function genericHandler(ResponseInterface $response)
     {
         $code = (string) $response->getStatusCode();
+        if ($queryError = $response->getHeaderLine('x-amzn-query-error')) {
```

Comment:
We can't be positive this header will only exist on the services that are switching their protocols and ideally would check the models for the `awsQueryCompatible` coral trait

Reply:
discussed offline. Will include the change in the next revision.